lib/internal/Magento/Framework/Stdlib/DateTime/Timezone.php

```diff
@@ -174,7 +183,13 @@ public function date($date = null, $locale = null, $useTimezone = true, $include
                     $timeType,
                     new \DateTimeZone($timezone)
                 );
+
+                //Know if the date contains time or not
+                if (!preg_match('/(.*).(\d{2}):(\d{2})/', $date)) {
```

Comment:
Looking at logic of this method I believe it should be rewritten into 

```
if ($includeTime && !preg_match('\d{2}:\d{2}', $date))
```

(regexp note: no need in brackets, first part seems to be useless)